### PR TITLE
Migrate to TS code

### PR DESCRIPTION
**Pull request by winkerVSbecks**
### What I did
* Migrated all code to TS
* Added an eject command to switch to JS

#9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
     "storybook": "start-storybook -p 6006",
     "start": "concurrently \"yarn build:watch\" \"yarn storybook -- --no-manager-cache --quiet\"",
     "build-storybook": "build-storybook",
-    "prerelease": "node check-metadata.js",
-    "release": "yarn build && auto shipit"
+    "prerelease": "node scripts/check-metadata.js",
+    "release": "yarn build && auto shipit",
+    "ejectTS": "zx scripts/eject-typescript.mjs"
```
Comment (shilman) [inline package.json:40]: maybe `eject-ts` to make it consistent?

File: scripts/eject-typescript.mjs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env zx
```
Comment (shilman) [inline scripts/eject-typescript.mjs:1]: whoa this is badass! 😍

File: package.json (hunk 1 above)
Comment (shilman) [inline package.json:40]: maybe `eject-ts` to make it consistent?

File: scripts/eject-typescript.mjs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// Copy TS files and delete src
+await $`cp -r ./src ./srcJS`;
```
Comment (shilman) [inline scripts/eject-typescript.mjs:4]: shouldn't this be `srcTS`?

File: src/Panel.tsx [hunk 4]
```diff
@@ -4,7 +4,11 @@ import { AddonPanel } from "@storybook/components";
 import { ADDON_ID, EVENTS } from "./constants";
 import { PanelContent } from "./components/PanelContent";
 
-export const Panel = (props) => {
+type PanelProps = {
```
Comment (shilman) [inline src/Panel.tsx:7]: maybe `interface`? i think TS people prefer them although I don't know the rationale. cc @ndelangen

File: src/Panel.tsx [hunk 5]
```diff
@@ -4,7 +4,11 @@ import { AddonPanel } from "@storybook/components";
...
+  active: boolean;
+};
+
+export const Panel = (props: PanelProps) => {
```
Comment (shilman) [inline src/Panel.tsx:11]:
```suggestion
export const Panel: FC<PanelProps> = (props) => {
```

And suggest the same for the others

**Review by shilman — COMMENTED** [review]
Awesome! Love the eject functionality -- so cool. Lots of small comments but nothing serious.

File: src/Panel.tsx (hunk 4 above)
Comment (winkerVSbecks) [reply src/Panel.tsx:7]: Looks like we use interface everywhere else. I'll use that.

**Comment by winkerVSbecks** [comment]
@shilman updated

**Pull request merged**